### PR TITLE
feat: history cleanup

### PR DESCRIPTION
**Pull request by darkskygit**
fix AFF-1069

### CODE REVIEW

**Comment by darkskygit** [comment]
* **#7007** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7007?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @darkskygit and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/schema.prisma [hunk 1]
```diff
@@ -480,12 +480,15 @@ model AiSessionMessage {
...
-  createdAt   DateTime @default(now()) @map("created_at") @db.Timestamptz(6)
+  id          String    @id @default(uuid()) @db.VarChar(36)
+  userId      String    @map("user_id") @db.VarChar(36)
+  workspaceId String    @map("workspace_id") @db.VarChar(36)
+  docId       String    @map("doc_id") @db.VarChar(36)
+  promptName  String    @map("prompt_name") @db.VarChar(32)
+  messageCost Int       @default(0)
+  tokenCost   Int       @default(0)
+  createdAt   DateTime  @default(now()) @map("created_at") @db.Timestamptz(6)
+  deletedAt   DateTime? @map("deleted_at") @db.Timestamptz(6)
```
Comment (forehalo) [inline packages/backend/server/schema.prisma:491]: delete it directly?

Comment (darkskygit) [reply packages/backend/server/schema.prisma:491]: we need keep session cost to statistics the usage, will use to limit quota

File: packages/backend/server/src/plugins/copilot/session.ts [hunk 2]
```diff
@@ -495,6 +500,35 @@ export class ChatSessionService {
...
+        })
+        .then(sessions => sessions.map(({ id }) => id));
+      // cleanup all messages
+      await tx.aiSessionMessage.deleteMany({
+        where: { sessionId: { in: sessions } },
+      });
+      // only mark action session as deleted
+      // chat session always can be reuse
+      await tx.aiSession.updateMany({
+        where: { id: { in: sessions }, prompt: { action: { not: null } } },
```
Comment (forehalo) [inline packages/backend/server/src/plugins/copilot/session.ts]: the join query would be super slow. how do you think bringing a little redundant data like action to ai sessions table

Comment (darkskygit) [reply packages/backend/server/src/plugins/copilot/session.ts]: now to use local prompt cache to check, this info basically not be modified, so it is safe enough to use cache first

**Review by forehalo — APPROVED** [review]

**Pull request merged**